### PR TITLE
Session & cookie extention handled in the middleware

### PR DESCRIPTION
**Pull request by hero101**
### Describe the background of your pull request

What does your pull request do? Does it solve a bug (which one?), add a feature?

### Additional context

Add any other context

### Governance 

- [ ] Documentation is added
- [ ] Test cases are added or updated

By submitting this pull request I confirm that:
- I've read the contributing document https://github.com/alkem-io/.github/blob/master/CONTRIBUTING.md.
- I've read and understand the Code of Conduct https://github.com/alkem-io/.github/blob/master/CODE_OF_CONDUCT.md.
- I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/alkem-io/Coordination/blob/master/LICENSE.
- I understand that submitting the pull request requires agreeing to and signing the contributor license agreement.

### CODE REVIEW

**Comment by Comoque1** [comment]
With the latest changes (tested on TEST env.) and 3 server pods:
- data is correctly loaded for unauthenticated user
- ory_kratos_session is updated with every request

File: src/core/authentication/authentication.service.ts [hunk 1]
```diff
@@ -136,31 +136,34 @@ export class AuthenticationService {
...
-  public async extendSession(sessionToBeExtended: Session): Promise<void> {
+  public async extendSession(sessionToBeExtended: Session): Promise<Session> {
     const bearerToken = await getBearerToken(
       this.kratosPublicUrlServer,
       this.adminPasswordIdentifier,
       this.adminPassword
     );
 
-    await this.tryExtendSession(sessionToBeExtended, bearerToken);
+    return this.tryExtendSession(sessionToBeExtended, bearerToken);
```
Comment (valentinyanakiev) [inline src/core/authentication/authentication.service.ts:146]: I've put this await here last time as errors weren't propagated / logged on the main thread - please check whether throwing a kratos error will be logged correctly if you don't await.

File: src/core/middleware/session.extend.middleware.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    const authorization = req.headers['authorization'];
+
+    if (!authorization) {
+      return next();
+    }
+
+    const [, token] = authorization.split(' ');
+
+    if (!token) {
+      return next();
```
Comment (valentinyanakiev) [inline src/core/middleware/session.extend.middleware.ts:38]: add verbose log before calling next()

File: src/core/middleware/session.extend.middleware.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    @Inject(WINSTON_MODULE_NEST_PROVIDER)
+    private readonly logger: LoggerService,
+    private readonly authService: AuthenticationService
+  ) {}
+
+  async use(req: Request, res: Response, next: NextFunction) {
+    const authorization = req.headers['authorization'];
+
+    if (!authorization) {
+      return next();
```
Comment (valentinyanakiev) [inline src/core/middleware/session.extend.middleware.ts:31]: add verbose log before calling next()

File: src/core/middleware/session.extend.middleware.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    const [, token] = authorization.split(' ');
+
+    if (!token) {
+      return next();
+    }
+
+    const { session } = jwt_decode<KratosPayload>(token);
+
+    if (!session) {
+      return next();
```
Comment (valentinyanakiev) [inline src/core/middleware/session.extend.middleware.ts:47]: add verbose log before calling next()

**Review by valentinyanakiev — CHANGES_REQUESTED** [review]

File: src/core/authentication/authentication.service.ts (hunk 1 above)
Comment (hero101) [reply src/core/authentication/authentication.service.ts:146]: `extendSession` has a `catch` where it's used

Comment (valentinyanakiev) [reply src/core/authentication/authentication.service.ts:146]: Tried it - works for me :)

**Review by valentinyanakiev — APPROVED** [review]

**Pull request merged**